packages/core/admin/admin/src/hooks/useRegenerate/index.js

```diff
@@ -18,10 +18,15 @@ const useRegenerate = (url, id, onRegenerate) => {
       onRegenerate(accessKey);
     } catch (error) {
       setIsLoadingConfirmation(false);
-      toggleNotification({
-        type: 'warning',
-        message: get(error, 'response.data.message', 'notification.error'),
-      });
+
+      if (onError) {
+        onError(error);
+      } else {
+        toggleNotification({
+          type: 'warning',
+          message: get(error, 'response.data.message', 'notification.error'),
```

Comment:
Could you please use the new `useAPIErrorHandler()` hook instead? https://contributor.strapi.io/hooks/use-api-error-handler

Reply:
yes sure